pkg/common/runner/service.go

```diff
@@ -47,8 +47,9 @@ func (r *Runner) createService(pod *kubev1.Pod) (svc *kubev1.Service, err error)
 // waitForCompletion will wait for a runner's pod to have a valid v1.Endpoint available
 func (r *Runner) waitForCompletion(podName string, timeoutInSeconds int) error {
 	var endpoints *kubev1.Endpoints
+	var err error
 	var pendingCount int = 0
-	return wait.PollImmediate(slowPoll, time.Duration(timeoutInSeconds)*time.Second, func() (done bool, err error) {
+	return wait.PollUntilContextTimeout(context.Background(), slowPoll, time.Duration(timeoutInSeconds)*time.Second, false, func(ctx context.Context) (bool, error) {
 		endpoints, err = r.Kube.CoreV1().Endpoints(r.svc.Namespace).Get(context.TODO(), r.svc.Name, metav1.GetOptions{})
```

Comment:
```suggestion
		endpoints, err = r.Kube.CoreV1().Endpoints(r.svc.Namespace).Get(ctx, r.svc.Name, metav1.GetOptions{})
```
we can use the provided context now

Reply:
There is one more instance of `context.TODO` below (line 63)

Reply:
One more here and LGTM! https://github.com/openshift/osde2e/pull/2435/files#diff-e408dafcba8a42fedcabe58b2b2192906fa703ab039b8f2e3d9b2df678406575R62

Thanks for updating these. It will be cool to refactor things further next to allow passing down the parent ginkgo context to all the functions and really take advantage of context cancellation/timeout!